tests/mod.rs

```diff
@@ -94,6 +94,25 @@ where
     // `FailurePersistence`s for each test
 }
 
+/// Validates that replay-from-seed works by running a failing seed found by a random scheduler for one
+/// iteration, expecting it to fail, comparing the new failing schedule against the previously collected one,
+/// and checking the two schedules being identical.
+fn check_replay_from_seed_match_schedule<F>(test_func: F, seed: u64, schedule: &str)
+where
+    F: Fn() + Send + Sync + UnwindSafe + 'static,
+{
+    let result = {
+        panic::catch_unwind(move || {
+            replay_random_from_seed(test_func, seed);
+        })
+        .expect_err("replay should panic")
+    };
+    let new_output = result.downcast::<String>().unwrap();
+    let new_schedule = parse_schedule::from_stdout(&new_output).expect("output should contain a schedule");
```

Comment:
nit: `new_schedule` is non-descriptive. Maybe `schedule_from_replay` ? Maybe rename `schedule` in the params to `expected_schedule` ?